### ISSUE TITLE
[BUG] BlobBaseClient.Exists always logs exceptions to App Insights when the blob does not exist.

### ISSUE THREAD

**Issue by mrpmorris** [Azure/azure-sdk-for-net#18592]
**Describe the bug**
BlobBaseClient.Exists files lots of false exceptions in App Insights when the blob does not exist.

**Expected behavior**
Checking if a blob exists should simply return false, not log an exception in App Insights.

**Actual behavior (include Exception or Stack Trace)**
App Insights is littered with false-positive exceptions due to a 404 response code to a HEAD request.

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

1. Create an app that connects to a valid blob storage
2. Enable AppInsights
3. Access BlobBaseClient.Exists for a blob you know does not exist
4. Look in AppInsights

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Blobs 12.4.4
 - Hosting platform or OS and .NET runtime version: Azure Function app (net core 3.1)

The problem seems to be here
https://github.com/Azure/azure-sdk-for-net/blob/f01cbdc6b939331123af0c682892f3f478350f27/sdk/storage/Azure.Storage.Queues/src/Generated/ServiceRestClient.cs#L162

The code throws an exception for anything other than a 200 response - but 404 is not an exceptional response, it is very much expected - in fact, I expect to call it for non-existent blobs more often than any other blob scenario.

Ultimately, the exception is swallowed here
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs#L2952

But, by this point in time, the exception has already been logged to App Insights.

**Comment by jsquire** [comment]
//cc: @tg-msft, as he was involved in an earlier discussion on this topic.

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by tg-msft** [comment]
This is waiting on #9934.

**Comment by janid1967** [comment]
I am experiencing the same problem and looking forward to a fix of this

**Comment by seanmcc-msft** [comment]
@tg-msft, what is the status of https://github.com/Azure/azure-sdk-for-net/issues/9934 ?

**Comment by tg-msft** [comment]
Last I heard, we're still officially waiting for an answer from the Observability folks.  +@pakrym who recently removed some noise for CreateIfNotExists calls and can comment further.  I don't think his recent change would help with BlobBaseClient.Exists though because we don't add any conditional headers to the request.

**Comment by pakrym** [comment]
We haven't gotten clear guidance but I think we can still make progress on our own. We are going to start with fixing the `CreateIfNotExists` and then look into adding support to pass ResponseClassifier to the code-generated methods. That way we can create and pass custom classifiers to calls that return expected 404s.

**Comment by amnguye** [comment]
@tg-msft @pakrym was there an update on #9934 yet ?
Also did CreateIfNotExists get fixed in regards to this issue?

**Comment by bruckGT1** [comment]
I am also seeing this behavior. Any updates?

**Comment by amishra-dev** [comment]
@pakrym Pavel will we be accepting the fix for #9934?

**Comment by tg-msft** [comment]
We don't want the fix at #9934 - we want to enable this via the work being done as part of https://github.com/Azure/azure-sdk-for-net/issues/25626 that will be the standard pattern across the Azure SDK for .NET.

**Comment by amishra-dev** [comment]
@tg-msft Ted any update on this? @jaschrep-msft can you please track this from our side?

**Comment by tg-msft** [comment]
Yes, the basic Azure.Core features are now in (but not shipped yet) and the generator has been updated to produce LLC protocol methods with a flag.  That means you could start enabling this with a package reference, though it won't fully light up until a fix for https://github.com/azure/autorest.csharp/issues/2006 is committed and deployed.

**Comment by older** [comment]
@tg-msft It looks like that AutoRest issue was fixed in March. Is this one going to be fixed as well?

**Comment by annelo-msft** [comment]
The required APIs should be available now in Azure.Core to be able to implement this feature.  In order to suppress distributed tracing for a given status code, `AddClassifier()` should be called on `RequestContext` prior to passing it to the method in question.  An example can be found in the TableServiceClient [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs#L697).

Please reach out with any remaining questions on implementation.
